discovery-provider/src/solana/anchor_program_indexer.py

```diff
@@ -162,13 +173,32 @@ def extract_ids(self, processed_transactions: List[ParsedTx]):
                 elif instruction_name == "update_admin":
                     pass
                 elif instruction_name == "init_user":
-                    # No action to be taken here
-                    pass
+                    user_id = instruction.get("data").get("user_id")
+                    user_storage_account = str(
+                        instruction.get("account_names_map").get("user")
+                    )
+                    user_storage_to_id[user_storage_account] = user_id
+                    entities["users"].add(user_id)
 
                 elif instruction_name == "init_user_sol":
-                    # TODO: parse the tx data for their user id and fetch
-                    # user_id = instruction.get("data").get("id")
-                    pass
+                    # Fetch user_id and embed in instruction
+                    user_storage_account = str(
+                        instruction.get("account_names_map").get("user")
+                    )
+                    if user_storage_account not in user_storage_to_id:
```

Comment:
i think we do this because we're assuming either the init_user tx is indexed or it's in this same batch. there doesn't need to be a check for the case were we get init_user_sol before init_user because the program would've rejected it, right?

Reply:
that's exactly right